src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -72,20 +72,29 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+        // We try to find the person based on the given studentId.
+        Optional<Person> personToEdit = lastShownList.stream()
+                .filter(person -> person.getSid() == index.getOneBased())
+                .findFirst();
+
+
+        if (personToEdit.isPresent()) {
+            Person editedPerson = createEditedPerson(personToEdit.get(), editPersonDescriptor);
+
+            // Checks if the new person is the same as the unedited person.
+            if (!personToEdit.get().isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
```

Comment:
Base on my testing this does not work.
For example 2 consecutive commands:

edit 2 n/Yuri
edit 2 n/Yuri

does not return in an Exception

Reply:
Same behaviour as AB3. solved!